File: go-sdk/pkg/models/operations/getcommentsbyarticleid.go
package operations

import (
	"github.com/speakeasy-sdks/forem-sdks/go-client-sdk/pkg/models/shared"
)

type GetCommentsByArticleIDQueryParams struct {
	AID *string `queryParam:"style=form,explode=true,name=a_id"`
	PID *string `queryParam:"style=form,explode=true,name=p_id"`
}

type GetCommentsByArticleIDRequest struct {
	QueryParams GetCommentsByArticleIDQueryParams
}

type GetCommentsByArticleIDResponse struct {
	Comments    []shared.Comment
	ContentType string
	StatusCode  int64
}

File: ts-sdk/src/sdk/models/operations/getcommentsbyarticleid.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetCommentsByArticleIdQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=a_id" })
  aId?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=p_id" })
  pId?: string;
}


export class GetCommentsByArticleIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetCommentsByArticleIdQueryParams;
}


export class GetCommentsByArticleIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata({ elemType: shared.Comment })
  comments?: shared.Comment[];

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
